src/hooks/useTokenList.ts

```diff
@@ -8,12 +8,15 @@ import { getTokens, subscribeToTokenList } from 'services'
 const emptyArray: TokenDetails[] = []
 
 export const useTokenList = (networkId?: number): TokenDetails[] => {
-  const [tokens, setTokens] = useSafeState(networkId === undefined ? emptyArray : getTokens(networkId))
+  // sync get tokenList
+  const tokens = networkId === undefined ? emptyArray : getTokens(networkId)
+
+  // force update with a new value each time
+  const [, forceUpdate] = useSafeState({})
 
   useEffect(() => {
-    if (networkId === undefined) return setTokens(emptyArray)
-    return subscribeToTokenList(setTokens)
-  }, [networkId, setTokens])
+    return subscribeToTokenList(() => forceUpdate({}))
```

Comment:
I don't get this.

The callback to the subscription is a function that updates an empty state of something that's ignored?

:confused:

Reply:
Just forces rerendet. seState(new value), and object is airways new, can use symbol too, or array, or counter +1

Reply:
Complicated, but now I got it :)

Reply:
Sorry, I don't get it either 🤔

On every change of the token list, we call `forceUpdate` wich it looks doesn't have an effect, because no-one uses his value. The value is ignored.

That `useEffect ` is a bit strange to me, depends on `forceUpdate` function, and that doesn't change for a component using this hook. It made more sense to me, the previous hook, where were we had: `subscribeToTokenList(setTokens)`, so I'm a bit lost.